docker/client.py

```diff
@@ -780,7 +780,7 @@ def start(self, container, binds=None, port_bindings=None, lxc_conf=None,
             publish_all_ports=publish_all_ports, links=links, dns=dns,
             privileged=privileged, dns_search=dns_search, cap_add=cap_add,
             cap_drop=cap_drop, volumes_from=volumes_from, devices=devices,
-            network_mode=network_mode, restart_policy=restart_policy,
+            network_mode=network_mode or '', restart_policy=restart_policy,
```

Comment:
OK, that makes sense. (I'd be in favour of removing `start`'s arguments soon.)

Could we make it clearer what's happening? Perhaps:

```python
start_config_kwargs = dict(
    binds=binds, port_bindings=port_bindings, lxc_conf=lxc_conf,
    publish_all_ports=publish_all_ports, links=links, dns=dns,
    privileged=privileged, dns_search=dns_search, cap_add=cap_add,
    cap_drop=cap_drop, volumes_from=volumes_from, devices=devices,
    network_mode=network_mode, restart_policy=restart_policy,
    extra_hosts=extra_hosts, read_only=read_only, pid_mode=pid_mode,
    ipc_mode=ipc_mode, security_opt=security_opt, ulimits=ulimits
)

start_config = None

if any(v is not None for v in start_config_kwargs.values()):
    start_config = create_host_config(**start_config_kwargs)
```